lib/run_loop.rb

```diff
@@ -301,4 +306,16 @@ def self.detect_automator(options, xcode, device)
       RunLoop.default_automator(xcode, device)
     end
   end
+
+  def self.keyboard_hidden?
+    File.file?(preferences_plist_path) && RunLoop::PlistBuddy.new.plist_read('HardwareKeyboardLastSeen', preferences_plist) == 'true'
+  end
+
+  def self.restore_keyboard
+    RunLoop::PlistBuddy.new.plist_set('HardwareKeyboardLastSeen', 'bool', 'NO', preferences_plist_path)
+  end
+
+  def self.preferences_plist_path
+    "#{device.simulator_root_dir}/data/Library/Preferences/com.apple.Preferences.plist"
```

Comment:
I think if you want to be a bit more nicer here, you could use the [Pathname](https://ruby-doc.org/stdlib-1.9.3/libdoc/pathname/rdoc/Pathname.html). And use the `#join` method to construct the path. It will then use `/` or `\` depending on the operating system.